backend/runner/taskrun/schedulerv2.go

```diff
@@ -173,27 +173,14 @@ func (s *SchedulerV2) scheduleAutoRolloutTask(ctx context.Context, taskUID int)
 		if plan == nil {
 			return true, nil
 		}
-		planCheckRuns, err := s.store.ListPlanCheckRuns(ctx, &store.FindPlanCheckRunMessage{PlanUID: &plan.UID})
+		latestRuns, err := s.store.ListPlanCheckRuns(ctx, &store.FindPlanCheckRunMessage{
+			PlanUID:    &plan.UID,
+			LatestOnly: true,
+		})
 		if err != nil {
-			return false, errors.Wrapf(err, "failed to list plan check runs")
-		}
-		type key struct {
-			instanceUID  int
-			databaseName string
-			checkType    store.PlanCheckRunType
-		}
-		latestRun := map[key]*store.PlanCheckRunMessage{}
-		for _, run := range planCheckRuns {
-			k := key{
-				instanceUID:  int(run.Config.InstanceUid),
-				databaseName: run.Config.DatabaseName,
-				checkType:    run.Type,
-			}
-			if latest, ok := latestRun[k]; !ok || latest.UID < run.UID {
-				latestRun[k] = run
-			}
+			return false, errors.Wrapf(err, "failed to list latest plan check runs")
 		}
-		for _, run := range latestRun {
+		for _, run := range latestRuns {
```

Comment:
Might want find a way to cancel all previous plan check runs. Not this CL though.